src/main/java/dev/codesupport/web/api/service/ArticleServiceImpl.java

```diff
@@ -0,0 +1,225 @@
+    }
+
+    @Override
+    public Article getArticleById(Long id) {
+        return crudOperations.getById(id);
+    }
+
+    @Override
+    public Article createArticle(Article article) {
+        return crudOperations.createEntity(article);
+    }
+
+    @Override
+    public Article updateArticle(Article article) {
+        return null;
+    }
+
+    @Override
+    public Article deleteArticle(Article article) {
+        return null;
+    }
+
+    @Override
+    public Article publishArticle(Article article) {
+        Optional<ArticleEntity> optionalArticle = articleRepository.findById(article.getId());
+
+        if (optionalArticle.isPresent()) {
+            PublishedArticleEntity publishedArticleEntity;
+            Optional<PublishedArticleEntity> optionalPA = publishedArticleRepository.findByArticleCode(optionalArticle.get().getArticleCode());
+
+            if (optionalPA.isPresent()) {
+                publishedArticleEntity = updatePublishedArticle(optionalPA.get(), optionalArticle.get());
+            } else {
+                publishedArticleEntity = createPublishedArticle(optionalArticle.get());
+            }
+
+            publishedArticleRepository.save(publishedArticleEntity);
+        } else {
+             throw new ResourceNotFoundException(ResourceNotFoundException.Reason.NOT_FOUND);
+        }
+        return null;
+    }
+
+    PublishedArticleEntity updatePublishedArticle(PublishedArticleEntity paEntity, ArticleEntity article) {
+        paEntity.setArticleId(article.getId());
+        return paEntity;
+    }
+
+    PublishedArticleEntity createPublishedArticle(ArticleEntity article) {
+        PublishedArticleEntity newPAEntity = new PublishedArticleEntity();
+        newPAEntity.setArticleId(article.getId());
+        newPAEntity.setArticleCode(article.getArticleCode());
+        newPAEntity.setPublished(true);
+        return newPAEntity;
+    }
+
+    static class CrudLogic {
+
+        private final ArticleRepository articleRepository;
+        private final TagRepository tagRepository;
+        private final TagSetRepository tagSetRepository;
+        private final TagSetToTagsRepository tagSetToTagsRepository;
+
+        private CrudLogic(
+                ArticleRepository articleRepository,
+                TagRepository tagRepository,
+                TagSetRepository tagSetRepository,
+                TagSetToTagsRepository tagSetToTagsRepository
+        ) {
+            this.articleRepository = articleRepository;
+            this.tagRepository = tagRepository;
+            this.tagSetRepository = tagSetRepository;
+            this.tagSetToTagsRepository = tagSetToTagsRepository;
+        }
+
+        Consumer<ArticleEntity> preSaveLogic() {
+            return (ArticleEntity articleEntity) -> {
+                updateTagReferences(articleEntity);
+                updateArticleMetaData(articleEntity);
+            };
+        }
+
+        Consumer<ArticleEntity> preGetLogic() {
```

Comment:
Codacy found an issue: [Use explicit scoping instead of the default package private level](https://app.codacy.com/gh/codesupport/website-backend/pullRequest?prid=6300789)